src/data/set/lattice.lean

```diff
@@ -1377,6 +1377,35 @@ begin
   { intros x hz x' hw, exact ⟨x ⊔ x', hs le_sup_left hz, ht le_sup_right hw⟩ }
 end
 
+lemma sInter_prod_sInter_subset (S : set (set α)) (T : set (set β)) :
+  ⋂₀ S ×ˢ ⋂₀ T ⊆ ⋂ r ∈ S ×ˢ T, r.1 ×ˢ r.2 :=
+subset_Inter₂ (λ x hx y hy, ⟨hy.1 x.1 hx.1, hy.2 x.2 hx.2⟩)
+
+lemma sInter_prod_sInter (S : set (set α)) (T : set (set β)) (hS : S.nonempty) (hT : T.nonempty) :
```

Comment:
```suggestion
lemma sInter_prod_sInter  {S : set (set α)} {T : set (set β)} (hS : S.nonempty) (hT : T.nonempty) :
```

Reply:
Presumably I should do the same for the other lemmas?

Reply:
I see you've done the others, thank you - except `sInter_prod_sInter_subset` - presumably because there are no additional hypothesis there from which `S` and `T` can be inferred?